### PR TITLE
[SUCE-354] Dynamic homepage data

### PR DESCRIPTION
**Pull request by gradam**

### CODE REVIEW

File: client/elements/addons/text-carusel.html [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+          ready() {
+            super.ready();
+            this._setInterval();
+            window.addEventListener('resize', e => {
+                this._debouncer = Polymer.Debouncer.debounce(
+                    this._debouncer,
+                    Polymer.Async.timeOut.after(200),
+                    () => {
+                        requestAnimationFrame(() => {
+                            this.$.container.style.height = 'unset'
```
Comment (TrebuhD) [inline client/elements/addons/text-carusel.html]: ;

File: client/elements/addons/text-carusel.html [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+                button.addEventListener('touchcancel', () => {
+                    this._removeActiveChevronClass(button, 100);
+                });
+                button.addEventListener('mouseout', e => {this._handleMouseOut(button, e)});
+            })
+          }
+
+          _handleMouseOut(button, e) {
+              const element = e.toElement || e.relatedTarget;
+              if (element.parentNode == button || element == button){
```
Comment (TrebuhD) [inline client/elements/addons/text-carusel.html]: please format this file in PyCharm

File: client/elements/addons/text-carusel.html [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+            }, timeout);
+          }
+
+          _setInterval() {
+                this.timer = setInterval(() => {
+                    this._nextItem();
+                }, 30000);
+          }
+
+          _nextItemClick(e) {
```
Comment (TrebuhD) [inline client/elements/addons/text-carusel.html]: `_loadNextItem`

File: client/elements/addons/text-carusel.html [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+          _nextItem() {
+              this.selectedItemIndex = (this.selectedItemIndex + 1) % this.elements.length;
+          }
+
+          _previousItem() {
+              if(this.selectedItemIndex === 0) {
+                  this.selectedItemIndex = this.elements.length - 1;
+              } else {
+                  this.selectedItemIndex = (this.selectedItemIndex - 1) % this.elements.length;
```
Comment (TrebuhD) [inline client/elements/addons/text-carusel.html]: no modulo necessary

File: client/elements/static/home-page.html [hunk 5]
```diff
@@ -342,9 +385,14 @@
...
                 <div class="card-content-text">
-                  <div class="card-text center-text">I understood: ‘Rebirth is ended; the spiritual journey has been completed; 
-                    what had to be done has been done; there is no return to any state of existence.’</div>
-                  <paper-button class="button horizontal-center">READ THIS SUTTA</paper-button>
+                  <template is="dom-if" if="[[loadingEpigraphs]]">
+                      <paper-spinner-lite active class="spinner"></paper-spinner-lite>
+                    </template>
+                    <template is="dom-if" if="[[!loadingEpigraphs]]">
+                    <div class="card-text center-text">
+                        <sc-text-carusel elements="[[epigraphsData]]" display="epigraph" link="uid"></sc-text-carusel>
```
Comment (TrebuhD) [inline client/elements/static/home-page.html]: carousel

File: client/elements/static/home-page.html [hunk 6]
```diff
@@ -373,8 +421,12 @@
                 and have tried to make it a little easier for you. Here’s some things you can try.</div>
                 </div>
             </paper-card>
-            <paper-button class="button first-card-button grid-button">INTRODUCTION</paper-button>
-            <paper-button class="button second-card-button grid-button">GET STARTED</paper-button>
+            <a class="button-link first-card-button grid-button" href="/introduction">
+              <paper-button class="button">INTRODUCTION</paper-button>
```
Comment (TrebuhD) [inline client/elements/static/home-page.html]: zero

Comment (gradam) [reply client/elements/static/home-page.html]: nope. it's O

File: client/elements/addons/text-carousel.html [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+                      value: true
+                  }
+              }
+          }
+
+          ready() {
+              super.ready();
+              this._setInterval();
+              this._resizeContainerOnWindowResize();
+              this._addChevronHighlihtListeners();
```
Comment (TrebuhD) [inline client/elements/addons/text-carousel.html]: Typo in highlight

File: client/elements/addons/text-carousel.html [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+                  });
+                  button.addEventListener('touchcancel', () => {
+                      this._removeActiveChevronClass(button, 100);
+                  });
+                  button.addEventListener('mouseout', e => {
+                      this._handleMouseOut(button, e)
+                  });
+              })
+          }
+
```
Comment (TrebuhD) [inline client/elements/addons/text-carousel.html]: Redundant blank lines

**Pull request merged**